### DISCUSSION TITLE
Explode function not spliting MultiLineString  to lineString at junction

### DISCUSSION THREAD

**Discussion by Abhigulve** [geopandas/geopandas#2704] [Q&A]
This is our code.
```
multi_line_string = gdf.boundary
boundry_lineString = gpd.GeoDataFrame([multi_line_string], crs="EPSG:4326",geometry=multi_line_string)
explode = boundry_lineString.explode()
```
The following is the result of this operation. It is not breaking at the junction.
This happens with some of the MultiLineString, not for all.

![MicrosoftTeams-image](https://user-images.githubusercontent.com/16499384/94808510-dc7a7c00-040e-11eb-816a-5bb8de17dfd8.png)

Expected result :
![MicrosoftTeams-image (1)](https://user-images.githubusercontent.com/16499384/94808513-de443f80-040e-11eb-92bb-c4e8555429af.png)

Please let us know, what we are doing wrong.

**Comment by brendan-ward** [comment]
@Abhigulve I think we need more information to be able to reproduce and investigate.  Can you please share this area of your geodataframe as a GIS file, GeoJSON, WKT, or something else that would let us exactly replicate your source geometry?

**Comment by Abhigulve** [comment]
hi @brendan-ward  please find used input file in attachment
[Source.zip](https://github.com/geopandas/geopandas/files/5312121/Source.zip)

**Comment by Abhigulve** [comment]
@martinfleis I have added the input source file. please let me know if anything else required

**Comment by brendan-ward** [comment]
@Abhigulve I think there is an error in your code; you are creating a GeoDataFrame with one column for each individual linestring, plus another column for `geometry`:

```
>>> boundry_lineString = gpd.GeoDataFrame([multi_line_string], crs="EPSG:4326",geometry=multi_line_string)
>>> boundry_lineString.columns
Index([         0,          1,          2,          3,          4,          5,
                6,          7,          8,          9,
       ...
              575,        576,        577,        578,        579,        580,
              581,        582,        583, 'geometry'],
      dtype='object', length=585)
```


I think you meant to create a GeoDataFrame of the linestrings as records, which then does not need to be exploded:
```
>>> boundry_lineString = gpd.GeoDataFrame(crs="EPSG:4326",geometry=multi_line_string)
>>> boundry_lineString
                                              geometry
0    LINESTRING (77.04091 31.22364, 77.04093 31.223...
1    LINESTRING (77.02107 31.18997, 77.01801 31.189...
2    LINESTRING (77.03649 31.20024, 77.04285 31.196...
3    LINESTRING (77.04798 31.20836, 77.04577 31.203...
4    LINESTRING (77.05398 31.19148, 77.05155 31.190...
..                                                 ...
579  LINESTRING (77.28388 31.12003, 77.28356 31.119...
580  LINESTRING (77.16103 31.12260, 77.15773 31.122...
581  LINESTRING (77.03575 31.16631, 77.03417 31.164...
582  LINESTRING (77.01681 31.18689, 77.01777 31.183...
583  LINESTRING (77.01777 31.18369, 77.02659 31.184...

[584 rows x 1 columns]
```

If you had other columns you wanted in that GeoDataFrame, you could use a `dict` in the constructor:

```
gpd.GeoDataFrame({'geometry': geometries, 'other_col': [...same length as geometries...]}, crs="EPSG:4326")
```

(Note: since `geometry` contains the geometries, it is automatically detected as the geometry column, otherwise you'd need to use the `geometry` parameter in the constructor to reference the name of the geometry column)

**Comment by Abhigulve** [comment]
Hi @brendan-ward  we have tried the code you provided 
```
multi_line_string = gdf.boundary
boundry_lineString = gpd.GeoDataFrame(crs="EPSG:4326",geometry=multi_line_string)
explode = boundry_lineString.explode()
```
The result looks like this : 

![MicrosoftTeams-image (5)](https://user-images.githubusercontent.com/16499384/94901484-57e23900-04b4-11eb-81de-57162260b2cc.png)

which is not matching with our requirement to break polylines at the junction.
One thing we observe that explode taking multipart geometry which we are created using the following code.
```
boundry_lineString = gpd.GeoDataFrame([multi_line_string], crs="EPSG:4326",geometry=multi_line_string)
explode = boundry_lineString.explode()
``` 
One more way is 
```
multi_line_string = gdf.boundary
boundry_lineString = gpd.GeoDataFrame(crs="EPSG:4326",geometry=multi_line_string)
Ml = boundry_lineString.intersection(boundry_lineString)
explode = boundry_lineString.explode()
```
This code does break lines on junction but also unnecessarily breaking withing the same polylines.

**Comment by martinfleis** [comment]
This is not a bug. You are expecting a behaviour for which the functions are not designed. To have a clean output you would need to identify junctions and extract edges between them. Ideally based on coordinates and some topology information.

The other way is using intersection between neighbouring polygons, but in that case, you would need to iterate over each pair (once, so A-B and not B-A).

If the only issue with your resulting geometry is that it is divided in some places where it should not be and nothing else, you can fix it using [momepy](http://docs.momepy.org/en/stable/generated/momepy.network_false_nodes.html#momepy.network_false_nodes). 

```python
import momepy

result = momepy.network_false_nodes(explode)
```

**Comment by Abhigulve** [comment]
@martinfleis  thanks for the suggestion. I'll try to use momepy
